config_ext_test.go

```diff
@@ -447,57 +418,210 @@ mismatch: [scalar]
+}
+
+func TestStrictYAML(t *testing.T) {
+	t.Skip("TODO: no strict mode support yet")
+	provider, err := NewYAMLProviderFromReader(
+		strings.NewReader(_base),
+		strings.NewReader(_override),
+	)
+	require.NoError(t, err, "couldn't create provider")
+	assert.Equal(t, `cached "yaml"`, provider.Get(Root).Source(), "wrong source")
+	assert.Equal(t, "<nil>", provider.Get("nothing").String(), "wrong string representation")
+
+	runCommonTests(t, provider)
+
+	t.Run("fail on type mismatch during merge", func(t *testing.T) {
+		_, err := NewYAMLProviderFromReader(
+			strings.NewReader("mismatch: foo"),   // scalar
+			strings.NewReader("mismatch: [foo]"), // sequence
+		)
+		require.Error(t, err, "couldn't create permissive provider with type mismatch")
+		assert.Contains(t, err.Error(), "couldn't merge", "unexpected error message")
+	})
+
+	t.Run("fail on type mismatches during WithDefault", func(t *testing.T) {
+		provider, err := NewYAMLProviderFromReader(
+			strings.NewReader("mismatch: foo"), // scalar
+		)
+		require.NoError(t, err, "couldn't create provider")
+
+		_, err = provider.Get("mismatch").WithDefault([]string{"foo"}) // sequence
+		assert.Error(t, err, "success on type mismatch in strict mode")
+		assert.Contains(t, err.Error(), "can't merge", "unexpected error message")
 	})
+
+	t.Run("fail on duplicate keys", func(t *testing.T) {
+		_, err := NewYAMLProviderFromReader(
+			strings.NewReader("dupe: foo\ndupe: bar"),
+		)
+		require.Error(t, err, "created strict provider with type mismatch")
+		assert.Contains(t, err.Error(), `key "dupe" already set in map`, "unexpected error message")
+	})
+
+	t.Run("fail on extra data during Populate", func(t *testing.T) {
+		provider, err := NewYAMLProviderFromReader(
+			strings.NewReader("foo: bar\nbaz: quux"),
+		)
+		require.NoError(t, err, "couldn't create provider")
+
+		c := struct {
+			Foo string
+		}{}
+		err = provider.Get(Root).Populate(&c)
+		require.Error(t, err, "populate succeeded")
+		assert.Contains(t, err.Error(), "field baz not found in type struct", "unexpected error message")
+	})
+
+	t.Run("must not provide ignored fields", func(t *testing.T) {
+		provider, err := NewYAMLProviderFromReader(strings.NewReader("toyota: camry"))
+		require.NoError(t, err, "couldn't create provider")
+
+		c := struct {
+			Toyota string `yaml:"-"`
+		}{}
+		err = provider.Get(Root).Populate(&c)
+		require.Error(t, err, "expected error")
+		assert.Contains(t, err.Error(), "field toyota not found in type", "unexpected error message")
+	})
+}
+
+func TestStaticFromYAML(t *testing.T) {
+	// Since we have a common test suite, we may as well use it to exercise the
+	// static provider too.
+	var base, override interface{}
+	require.NoError(t, yaml.Unmarshal([]byte(_base), &base), "couldn't unmarshal base YAML")
+	require.NoError(t, yaml.Unmarshal([]byte(_override), &override), "couldn't unmarshal base YAML")
+
+	t.Skip("TODO: no support for multiple sources in NewStaticProvider")
+	p := NopProvider{}
```

Comment:
Will this change to `NewStaticProvider` once we fix the test?

Reply:
Sort of - it'll change to a `NewYAML` with two different sources of static configuration.